core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/tenant/JDBCTenantManager.java

```diff
@@ -1274,12 +1276,23 @@ private StringBuilder printCurrentStackTrace() {
      * @return number of tenant count.
      * @throws UserStoreException Error when getting count of tenants.
      */
-    private int getCountOfTenants() throws UserStoreException {
+    private int getCountOfTenants(String domainName) throws UserStoreException {
 
-        String sqlStmt = TenantConstants.LIST_TENANTS_COUNT_SQL;
+        String sqlFilter = StringUtils.EMPTY;
+        if (StringUtils.isNotBlank(domainName)) {
```

Comment:
No need to validate any other special characters ? I believe better validate using a regex whether an acceptable domain name string.

Reply:
Will be fixed by https://github.com/wso2/identity-api-server/pull/723